app/models/host.rb

```diff
@@ -388,91 +352,74 @@ def start
-        check_policy_prevent("request_host_standby", "vim_power_down_to_standby")
-      else
-        _log.warn("Cannot go into standby mode from power state = <#{power_state.inspect}>")
-      end
+    if supports?(:standby)
+      check_policy_prevent("request_host_standby", "vim_power_down_to_standby")
     else
-      _log.warn("Cannot go into standby mode because <#{msg[:message]}>")
+      _log.warn("Cannot go into standby mode because <#{unsupported_reason(:standby)}>")
     end
   end
 
   def enter_maint_mode
-    msg = validate_enter_maint_mode
-    if msg[:available]
-      if power_state == 'on' && respond_to?(:vim_enter_maintenance_mode)
-        check_policy_prevent("request_host_enter_maintenance_mode", "vim_enter_maintenance_mode")
-      else
-        _log.warn("Cannot enter maintenance mode from power state = <#{power_state.inspect}>")
-      end
+    if supports?(:enter_maint_mode)
+      check_policy_prevent("request_host_enter_maintenance_mode", "vim_enter_maintenance_mode")
     else
-      _log.warn("Cannot enter maintenance mode because <#{msg[:message]}>")
+      _log.warn("Cannot enter maintenance mode because <#{unsupported_reason(:enter_maint_mode)}>")
     end
   end
 
   def exit_maint_mode
-    msg = validate_exit_maint_mode
-    if msg[:available] && respond_to?(:vim_exit_maintenance_mode)
+    if supports?(:exit_maint_mode)
       check_policy_prevent("request_host_exit_maintenance_mode", "vim_exit_maintenance_mode")
     else
-      _log.warn("Cannot exit maintenance mode because <#{msg[:message]}>")
+      _log.warn("Cannot exit maintenance mode because <#{unsupported_reason(:exit_maint_mode)}>")
     end
   end
 
   def shutdown
-    msg = validate_shutdown
-    if msg[:available] && respond_to?(:vim_shutdown)
+    if supports?(:shutdown)
       check_policy_prevent("request_host_shutdown", "vim_shutdown")
     else
-      _log.warn("Cannot shutdown because <#{msg[:message]}>")
+      _log.warn("Cannot shutdown because <#{unsupported_reason(:shutdown)}>")
     end
   end
 
   def reboot
-    msg = validate_reboot
-    if msg[:available] && respond_to?(:vim_reboot)
+    if supports?(:reboot)
       check_policy_prevent("request_host_reboot", "vim_reboot")
     else
-      _log.warn("Cannot reboot because <#{msg[:message]}>")
+      _log.warn("Cannot reboot because <#{unsupported_reason(:reboot)}>")
     end
   end
 
   def enable_vmotion
-    msg = validate_enable_vmotion
-    if msg[:available] && respond_to?(:vim_enable_vmotion)
+    if supports?(:enable_vmotion)
       check_policy_prevent("request_host_enable_vmotion", "vim_enable_vmotion")
     else
-      _log.warn("Cannot enable vmotion because <#{msg[:message]}>")
+      _log.warn("Cannot enable vmotion because <#{unsupported_reason(:enable_vmotion)}>")
     end
   end
 
   def disable_vmotion
-    msg = validate_disable_vmotion
-    if msg[:available] && respond_to?(:vim_disable_vmotion)
+    if supports?(:disable_vmotion)
```

Comment:
these feel like a common pattern

```ruby
  def warning_supports?(feature)
    if supports?(feature)
      true
    else
      _log.warn("Cannot #{feature} because <#{unsupported_reason(feature)}>")
      false
    end
  end

  if warning_supports?(:disable_vmotion)
    check_policy_prevent("request_host_disable_vmotion", "vim_disable_vmotion")
  end
```

Also wonder if moving the supports into `check_policy_prevent` would make sense.

Yes, feature will have an extra _ for one or two, but it is mostly consistent with the existing output.

It looks like `check_policy_prevent` is called from a few locations. Kinda seems like that should be a private method to me - but that may be because I'm suggesting changing the interface of it.

(FWIW/ I came up with quite a few incarnations of this method. So just pointing this out and I am not married to this suggestion)

Reply:
maybe `verbose_supports`

Reply:
Possibly?  But I think more likely is that we'd drop the supports check in the method itself, we don't do this anywhere else it is expected that the caller checks supports before invoking the method.